spatialdb/object.py

```diff
@@ -360,13 +360,22 @@ def add_cuboid_to_index(self, object_key, version=0, ingest_job=0):
         # range key is exp&ch&res&task
         ingest_job_range = "{}&{}&{}&{}".format(parts.experiment_id, parts.channel_id, parts.resolution, ingest_job)
 
+        # Partial lookup key stored so we can use a Dynamo query to find all cuboids
+        # tha belong to a channel.
+        lookup_key = '{}&{}&{}&{}'.format(
+            parts.collection_id, parts.experiment_id, parts.channel_id,
+            parts.resolution)
+
         try:
             dynamodb.put_item(
                 TableName=self.config['s3_index_table'],
-                Item={'object-key': {'S': object_key},
+                Item={
+                    'object-key': {'S': object_key},
                       'version-node': {'N': "{}".format(version)},
                       'ingest-job-hash': {'S': "{}".format(parts.collection_id)},
-                      'ingest-job-range': {'S': ingest_job_range}},
+                      'ingest-job-range': {'S': ingest_job_range},
+                      'lookup-key': {'S': lookup_key}
```

Comment:
In the comments / pull request you call this a partial lookup key, but in DynamoDB you call it a lookup key...

Reply:
No code currently uses this column. When we start indexing post-ingest, this will make finding all the cuboids that belong to a channel more efficient. I put this up now so we can potentially have this column populated before the next ingest.

I thought `lookup-key` was sufficient for the column name. I'm creating markdown files to sit along side the .json schema files that explain the "columns" used for each Dynamo item. Those will go in another PR/commit.

Reply:
This is the markdown I currently have that describes the s3 index:

# S3 Index

This Dynamo table tracks which cuboids exist in S3 cuboid bucket.  If the
cuboid belongs to an annotation channel, it also has a set that contains all
the annotation ids that exist inside the cuboid.

## Key

`object-key` (S): same key used in the S3 cuboid bucket
`version-node` (N): range key (reserved for future use)

## Attributes

`ingest-job-hash` (S): the collection id
`ingest-job-range` (S): the id of the ingest job that this cuboid was part of
`id-set` (SS): Set of all annotation ids that exist in the cuboid if this is part of an annotation channel
`lookup-key` (S): stores the collection | experiment | channel | resolution of the cuboid


## Global Secondary Indexes

### ingest-job-index

Find cuboids based on the ingest job.

#### Key

`ingest-job-hash`
`ingest-job-range` (range)

#### Projected Attributes

None


### project-index

Find all cuboids in a collection, experiment, channel at a resolution level.

##### Key

`lookup-key`

#### Projected Attributes

None